FieldOpt/Hdf5SummaryReader/hdf5_summary_reader.h

```diff
@@ -5,33 +5,76 @@
- * whereas if you get them using the get_*_injection_rates methods, they will be positive.
+ * Note also that when you get the rates for an injector using
+ * the get_*_rates methods, the numbers will be negative, whereas
+ * if you get them using the get_*_injection_rates methods, they
+ * will be positive.
+ *
+ * \todo This must also be tested for a 3 phase black oil model,
+ * it has only been tested for 2 phase dead oil.
+ *
+ * \todo Overall, Hdf5SummaryReader::readSaturation needs a better 
+ * implementation that reads soil and sgas at the same time/as a 
+ * single chunck from the H5 group, instead of reading from H5 one 
+ * vector at a time (the thinking is that the former is probably
+ * more efficient that the latter).
+ * In either case, the reading functionality needs to be flexible/
+ * /robust with respect to the different phase combinations that 
+ * might exist in the H5 group (currently GRIDPROPTIME), e.g., 
+ * soil/sgas, soil/sgas/swat, soil/swat, etc.
+ *
+ * \todo To make thing much tidier, collect variables containing 
+ * information about the reservoir cell ensemble, i.e., 
+ * cells_all_vector_, cells_active_, cells_inactive_, etc, and of
+ * reservoir cell data, i.e., pressure_, soil_, sgas_, swat_, into 
+ * one or more reservoir cells/cell data structs.
+ *
+ * \todo Later: to reduce the size of the H5 file, make ADGPRS/
+ * Optimizer save the saturation data (and whatever else) as 
+ * additional columns into the existing PTZ group (possibly 
+ * replacing current mobility columns), and remove GRIDPROPTIME 
+ * completely. 
+ * Additionally, which cell data types (including custom data 
+ * types) to print to H5 during simulation may be controlled by 
+ * custom simulator KEYWORD.
  *
- * \todo This must also be tested for a 3 phase black oil model, it has only been tested for 2 phase dead oil.
  */
 class Hdf5SummaryReader {
 public:
     /*!
-     * Read the HDF5 summary file written by AD-GPRS at the specified path.
+     * Read the HDF5 summary file written 
+     * by AD-GPRS at the specified path.
      *
-     * \note The provided path is not checked by this method, and should therefore be
-     * checked before invoking this.
+     * \note The provided path is not checked by this method,
+     * and should therefore be checked before invoking this.
      * @param file_path Path to a .H5 summary file.
-     * @return A Hdf5SummaryReader object containing information from the summary.
+     * @param cell_data_STATUS Flag for whether to read cell data
+     * or not (defaults to empty string)
+     * @param debug Flag to print H5 related data during testing
+     * (defaults to false)
+     * @return A Hdf5SummaryReader object containing 
+     * information from the summary.
      */
-    Hdf5SummaryReader(const std::string file_path);
+    Hdf5SummaryReader(const std::string file_path,
+                      std::string cell_data_STATUS = "",
```

Comment:
An even easier way to do this would be to replace this string with
`bool get_cell_data=false`
and then check in the implementation
```if (get_cell_data){
    ....
}
```

Your implementation is more flexible though, could allow for several different switches (e.g. get only some or all data). Thoughts?